vortex-array/src/array/chunked/compute/take.rs

```diff
@@ -88,8 +89,22 @@ fn take_strict_sorted(chunked: &ChunkedArray, indices: &Array) -> VortexResult<A
         // Now we can say the slice of indices belonging to this chunk is [pos, chunk_end_pos)
         let chunk_indices = slice(indices, pos, chunk_end_pos)?;
 
-        // Adjust the indices so they're relative to the chunk
-        let chunk_indices = subtract_scalar(&chunk_indices, &chunk_begin.into())?;
+        // Indices might not have a dtype big enough to fit chunk_begin after cast,
+        // if it does cast the scalar otherwise upcast the indices.
+        let chunk_indices = if chunk_begin < PType::try_from(chunk_indices.dtype())?.max_value() {
+            subtract_scalar(
+                &chunk_indices,
+                &Scalar::from(chunk_begin).cast(chunk_indices.dtype())?,
+            )?
+        } else {
+            // TODO: this is unnecessary, could instead upcast in the subtract.
```

Comment:
i think it's better to assert on it here, b/c you can do subtract_scalar with negative value. let's remove the TODO?

Reply:
will update comment